src/overlays/actors/ovl_En_Goma/z_en_goma.c

```diff
@@ -21,77 +50,836 @@ const ActorInit En_Goma_InitVars = {
+    if ((s8)this->actor.colChkInfo.health <= 0) {
+        this->unk_2CC = 5;
+        func_80032C7C(globalCtx, &this->actor);
+    } else {
+        this->unk_2CC = 10;
+    }
+
+    this->actor.speedXZ = 20.0f;
+    this->actor.posRot.rot.y = (this->actor.yawTowardsLink + 0x8000);
+    if (this->actor.params < 6) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_DAM1);
+    } else {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_DAM1);
+    }
+}
+
+void func_80A49AA8(EnGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->actor.bgCheckFlags & 1) {
+        Math_SmoothDownscaleMaxF(&this->actor.speedXZ, 1.0f, 2.0f);
+    }
+
+    if (this->unk_2CC == 0) {
+        if ((s8)this->actor.colChkInfo.health <= 0) {
+            func_80A49B30(this);
+        } else {
+            func_80A49294(this);
+        }
+    }
+}
+
+void func_80A49B30(EnGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000B78, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06000B78.genericHeader),
+                         2, -2.0f);
+    this->actionFunc = func_80A49BF0;
+    this->unk_2CC = 0x1E;
+    if (this->actor.params < 6) {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_DEAD);
+    } else {
+        Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_DEAD);
+    }
+
+    this->unk_2CA = 0x64;
+    this->actor.flags &= ~1;
+}
+
+void func_80A49BF0(EnGoma* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    if (this->actor.bgCheckFlags & 1) {
+        Math_SmoothDownscaleMaxF(&this->actor.speedXZ, 1.0f, 2.0f);
+    }
+
+    if (this->unk_2CC == 0x11) {
+        if (this->actor.params < 6) {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_BJR_LAND);
+        } else {
+            Audio_PlayActorSound2(&this->actor, NA_SE_EN_GOMA_JR_LAND);
+        }
+    }
+
+    if (this->unk_2CC == 0) {
+        func_80A49C94(this);
+    }
+}
+
+void func_80A49C94(EnGoma* this) {
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06000334, 1.0f, 0.0f, SkelAnime_GetFrameCount(&D_06000334.genericHeader),
+                         0, -2.0f);
+    this->actionFunc = func_80A49D0C;
+    this->unk_2CC = 3;
+}
+
+void func_80A49D0C(EnGoma* this, GlobalContext* globalCtx) {
+    Vec3f a;
+    Vec3f b;
+    BossGoma* goma;
+
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+    Math_SmoothDownscaleMaxF(&this->actor.speedXZ, 1.0f, 2.0f);
+
+    if (this->unk_2CC == 2) {
+        b.x = this->actor.posRot.pos.x;
+        b.y = (this->actor.posRot.pos.y + 5.0f) - 10.0f;
+        b.z = this->actor.posRot.pos.z;
+        a = D_80A4B7FC;
+        a.y = 0.03f;
+        func_80029DBC(globalCtx, &b, &D_80A4B7FC, &a, 0x28, 0);
+    }
+
+    if ((this->unk_2CC == 0) &&
+        (Math_SmoothScaleMaxMinF(&this->actor.scale.y, 0.0f, 0.5f, 0.00225f, 0.00001f) <= 0.001f)) {
+        if (this->actor.params < 6) {
+            // Warning, out of bounds
```

Comment:
what is this comment for? theres no string here

Reply:
Because I type unk_1A4 typed as an array of 4, the check is if params is < 6, so unk_1A4[4] and unk_1A4[5] is out of bounds.